ITGlueAPI/Internal/APICalls.ps1

```diff
@@ -0,0 +1,159 @@
+function New-ITGlue {
+    Param (
+        [Parameter(Mandatory = $true)]
+        [String]$resource_uri,
+        [Parameter(Mandatory = $true)]
+        [Hashtable]$data
+    )
+
+    $body = @{'data'= $data}
+    $body = ConvertTo-Json -InputObject $body -Depth $ITGlue_JSON_Conversion_Depth
+
+    try {
+        $ITGlue_Headers.Add('x-api-key', (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'N/A', $ITGlue_API_Key).GetNetworkCredential().Password)
+
+        $parameters = @{
+            "Method" = 'POST'
+            "Uri" = $ITGlue_Base_URI + $resource_uri
+            "Headers" = $ITGlue_Headers
+            "Body" = $body
+        }
+
+        $rest_output = Invoke-RestMethod @parameters -ErrorAction Stop
+    } catch {
+        Write-Error $_
+    } finally {
+        [void] ($ITGlue_Headers.Remove('x-api-key')) # Quietly clean up scope so the API key doesn't persist
+    }
+
+    $data = @{}
+    $data = $rest_output
+    return $data
+}
+
+function Get-ITGlue {
+    Param (
+        [Parameter(Mandatory = $true)]
+        [String]$resource_uri,
+
+        [Hashtable]$filter_list = @{}
+    )
+
+    $filter = ""
+    foreach($key in $filter_list.keys) {
+        $filter += "$key=$($filter[$key])&"
+    }
+
+    $filter = $filter.trimEnd("&") # Remove trailing &.
+
+    if(-not [System.String]::IsNullOrEmpty($filter)) { # Skip if no filer was added.
+        $resource_uri = "{0}?{1}" -f $resource_uri, $filter
+    }
+
+    try {
+        $ITGlue_Headers.Add('x-api-key', (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'N/A', $ITGlue_API_Key).GetNetworkCredential().Password)
+
+        $parameters = @{
+            "Method" = 'GET'
+            "Uri" = $ITGlue_Base_URI + $resource_uri
+            "Headers" = $ITGlue_Headers
+        }
+
```

Comment:
There's still a lot of code duplication across the four API call functions. I think they should be merged into a single `Invoke-APIRequest` (or similar) function. Here's my suggestion:

```powershell
function ConvertTo-QueryString([Hashtable]$url_params) {
    if ($null -eq $url_params) {
        return ""
    }

    $params = @()
    foreach ($key in $url_params.Keys) {
        $params += "$key=$($url_params[$key])"
    }

    $query_string = '?' + ($params -join '&')
    return $query_string
}

function Invoke-APIRequest {
    param (
        [ValidateSet('GET', 'POST', 'PATCH', 'DELETE')]
        [string]$method = 'GET',

        [Parameter(Mandatory = $true)]
        [string]$resource_uri,

        [Hashtable]$url_params = $null,

        [Hashtable]$data = $null
    )

    $query_string = ConvertTo-QueryString -url_params $url_params

    if ($null -eq $data) {
        $body = $null
    } else {
        $body = @{'data'=$data} | ConvertTo-Json -Depth $ITGlue_JSON_Conversion_Depth
    }

    try {
        $ITGlue_Headers.Add('x-api-key', (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'N/A', $ITGlue_API_Key).GetNetworkCredential().Password)

        $parameters = @{
            "Method" = $method
            "Uri" = $ITGlue_Base_URI + $resource_uri + $query_string
            "Headers" = $ITGlue_Headers
            "Body" = $body
        }

        $api_response = Invoke-RestMethod @parameters -ErrorAction Stop
    } catch {
        Write-Error $_
    } finally {
        [void] ($ITGlue_Headers.Remove('x-api-key')) # Quietly clean up scope so the API key doesn't persist
    }

    return $api_response
}
```

Of course, the calls to this function in the resource files would all need to be updated accordingly.